plugins/processors/gce/gce.go

```diff
@@ -0,0 +1,184 @@
+package ec2
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"cloud.google.com/go/compute/metadata"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/common/parallel"
+	"github.com/influxdata/telegraf/plugins/processors"
+)
+
+type GceProcessor struct {
+	AllowTags        []string        `toml:"allow_tags"`
+	Timeout          config.Duration `toml:"timeout"`
+	Ordered          bool            `toml:"ordered"`
+	MaxParallelCalls int             `toml:"max_parallel_calls"`
+	Log              telegraf.Logger `toml:"-"`
+
+	gceClient    *metadata.Client
+	allowTagsMap map[string]struct{}
+	parallel     parallel.Parallel
+	instanceID   string
+}
+
+const sampleConfig = `
+  ## GCP Instance and project metadata to attach to metrics as tags.
+  ## For more information see:
+  ## https://cloud.google.com/compute/docs/metadata/default-metadata-values
+  ##
+  ## Available tags:
+  ## * zone
+  ## * tags
+  ## * name
+  ## * hostname
+  allow_tags = []
+
+  ## Timeout for http requests made by against gce metadata endpoint.
+  timeout = "10s"
+
+  ## ordered controls whether or not the metrics need to stay in the same order
+  ## this plugin received them in. If false, this plugin will change the order
+  ## with requests hitting cached results moving through immediately and not
+  ## waiting on slower lookups. This may cause issues for you if you are
+  ## depending on the order of metrics staying the same. If so, set this to true.
+  ## Keeping the metrics ordered may be slightly slower.
+  ordered = false
+
+  ## max_parallel_calls is the maximum number of AWS API calls to be in flight
```

Comment:
```suggestion
  ## max_parallel_calls is the maximum number of GCP API calls to be in flight
```
What's the scenario in which multiple in-flight calls to the metrics endpoint will be useful? Do we hit it every time a metric is collected?

Reply:
It's a good question, I don't actually expect the labels to change much, how do you feel about them just being fetched once at startup and cached?

Here I was just mirroring the functionality from the AWS plugin.

Reply:
I have a feeling the pipeline apps just do a single fetch at startup so I dare say that would be fine.